internal/transformers/estransforms/optionalchain.go

```diff
@@ -11,7 +11,222 @@ type optionalChainTransformer struct {
+			if ast.IsSyntheticReferenceExpression(expression) {
+				res := ch.Factory().NewFunctionCallCall(expression.AsSyntheticReferenceExpression().Expression, expression.AsSyntheticReferenceExpression().ThisArg, args.Nodes)
+				res.Loc = node.Loc
+				ch.EmitContext().SetOriginal(res, node.AsNode())
+				return res
+			}
+			return ch.Factory().UpdateCallExpression(node, expression, nil, nil, args)
+		}
+	}
+	return ch.Visitor().VisitEachChild(node.AsNode())
+}
+
+func (ch *optionalChainTransformer) visitParenthesizedExpression(node *ast.ParenthesizedExpression, captureThisArg bool, isDelete bool) *ast.Node {
+	expr := ch.visitNonOptionalExpression(node.Expression, captureThisArg, isDelete)
+	if ast.IsSyntheticReferenceExpression(expr) {
+		// `(a.b)` -> { expression `((_a = a).b)`, thisArg: `_a` }
+		// `(a[b])` -> { expression `((_a = a)[b])`, thisArg: `_a` }
+		synth := expr.AsSyntheticReferenceExpression()
+		res := ch.Factory().NewSyntheticReferenceExpression(ch.Factory().UpdateParenthesizedExpression(node, synth.Expression), synth.ThisArg)
+		ch.EmitContext().SetOriginal(res, node.AsNode())
+		return res
+	}
+	return ch.Factory().UpdateParenthesizedExpression(node, expr)
+}
+
+func (ch *optionalChainTransformer) visitPropertyOrElementAccessExpression(node *ast.Expression, captureThisArg bool, isDelete bool) *ast.Expression {
+	if node.Flags&ast.NodeFlagsOptionalChain != 0 {
+		// If `node` is an optional chain, then it is the outermost chain of an optional expression.
+		return ch.visitOptionalExpression(node.AsNode(), captureThisArg, isDelete)
+	}
+	expression := ch.Visitor().VisitNode(node.Expression())
+	// Debug.assertNotNode(expression, isSyntheticReference); // !!!
+
+	var thisArg *ast.Expression
+	if captureThisArg {
+		if !transformers.IsSimpleCopiableExpression(expression) {
+			thisArg = ch.Factory().NewTempVariable()
+			ch.EmitContext().AddVariableDeclaration(thisArg)
+			expression = ch.Factory().NewAssignmentExpression(thisArg, expression)
+		} else {
+			thisArg = expression
+		}
+	}
+
+	if node.Kind == ast.KindPropertyAccessExpression {
+		p := node.AsPropertyAccessExpression()
+		expression = ch.Factory().UpdatePropertyAccessExpression(p, expression, nil, ch.Visitor().VisitNode(p.Name()))
+	} else {
+		p := node.AsElementAccessExpression()
+		expression = ch.Factory().UpdateElementAccessExpression(p, expression, nil, ch.Visitor().VisitNode(p.AsElementAccessExpression().ArgumentExpression))
+	}
+
+	if thisArg != nil {
+		res := ch.Factory().NewSyntheticReferenceExpression(expression, thisArg)
+		ch.EmitContext().SetOriginal(res, node.AsNode())
+		return res
+	}
+	return expression
+}
+
+func (ch *optionalChainTransformer) visitDeleteExpression(node *ast.DeleteExpression) *ast.Node {
+	unwrapped := ast.SkipParentheses(node.Expression)
+	if unwrapped.Flags&ast.NodeFlagsOptionalChain != 0 {
+		return ch.visitNonOptionalExpression(node.Expression, false, true)
+	}
+	return ch.Visitor().VisitEachChild(node.AsNode())
```

Comment:
Any reason you do this instead of the original-ish

```suggestion
	deletedExpr := ch.Visitor().VisitNode(node.Expression)
	return ch.Factory().UpdateDeleteExpression(node, deletedExpr)
```

Reply:
True, though it's technically doing a bit more work under the hood.